mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -180,75 +177,71 @@ impl AuthenticationSettings {
 
 #[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct CredentialProviderSettings {
+    #[serde(rename = "iothubhostname")]
+    iothub_hostname: String,
+
     #[serde(rename = "gatewayhostname")]
-    gateway_hostname: Option<String>,
+    gateway_hostname: String,
 
     #[serde(rename = "deviceid")]
-    device_id: Option<String>,
+    device_id: String,
 
     #[serde(rename = "moduleid")]
-    module_id: Option<String>,
+    module_id: String,
 
     #[serde(rename = "modulegenerationid")]
-    generation_id: Option<String>,
+    generation_id: String,
 
     #[serde(rename = "workloaduri")]
-    workload_uri: Option<String>,
+    workload_uri: String,
 }
 
 impl CredentialProviderSettings {
-    pub fn gateway_hostname(&self) -> Option<&str> {
-        self.gateway_hostname.as_ref().map(AsRef::as_ref)
+    pub fn iothub_hostname(&self) -> &str {
+        &self.iothub_hostname
     }
 
-    pub fn device_id(&self) -> Option<&str> {
-        self.device_id.as_ref().map(AsRef::as_ref)
+    pub fn gateway_hostname(&self) -> &str {
+        &self.gateway_hostname
     }
 
-    pub fn module_id(&self) -> Option<&str> {
-        self.module_id.as_ref().map(AsRef::as_ref)
+    pub fn device_id(&self) -> &str {
+        &self.device_id
     }
 
-    pub fn generation_id(&self) -> Option<&str> {
-        self.generation_id.as_ref().map(AsRef::as_ref)
+    pub fn module_id(&self) -> &str {
+        &self.module_id
     }
 
-    pub fn workload_uri(&self) -> Option<&str> {
-        self.workload_uri.as_ref().map(AsRef::as_ref)
+    pub fn generation_id(&self) -> &str {
+        &self.generation_id
     }
-}
-
-#[derive(Debug, Clone, Default, PartialEq, Deserialize)]
-pub struct Subscription {
-    pattern: String,
 
-    remote: String,
-}
-
-impl Subscription {
-    pub fn pattern(&self) -> &str {
-        &self.pattern
-    }
-
-    pub fn remote(&self) -> &str {
-        &self.remote
+    pub fn workload_uri(&self) -> &str {
+        &self.workload_uri
     }
 }
 
 #[derive(Debug, Clone, Default, PartialEq, Deserialize)]
-pub struct Forward {
+pub struct Topic {
     pattern: String,
 
-    remote: String,
+    local: Option<String>,
+
+    remote: Option<String>,
```

Comment:
Anca and I discussed offline and we think it should be:

```
"pattern": "temp/#",
"forward-prefix": "floor/kitchen/",
"subscribe-prefix": "local"
```

Reply:
will leave it local/remote for now, until we finalize the config schema